audit-checklists/Arbitrum.md

```diff
@@ -0,0 +1,95 @@
+# Arbitrum security checklist
+
+## General
+
+### Block number
+
+Invoking `block.number` in a smart contract on Arbitrum will return **L1** block number at which sequencer received transaction, not an **L2** block number. Also worth noting, block number being returned is not necessarily the latest L1 block number. It is the latest synced block number and syncing process occurs approximately every minute. In that period there can be ~5 new L1 blocks. So values returned by `block.number` on Arbitrum do not increase continuously but in "jumps" of ~5 blocks.
+
+> Always check for incorrect use of `block.number` on L2s, especially if it is used to track time over short periods.
+
+Bug examples: [1](https://code4rena.com/reports/2022-12-tigris#m-15-_checkdelay-will-not-work-properly-for-arbitrum-or-optimism-due-to-blocknumber-), [2](https://solodit.cyfrin.io/issues/incorrect-use-of-l1-blocknumber-on-arbitrum-cantina-none-uniswap-pdf)
+
```

Comment:
I see the recommendation in the Uniswap audit says -- Consider using `ArbSys(100).arbBlockNumber()` to retrieve the L2 block number on
Arbitrum. Is this valid? Do you think we can add it somewhere?

Reply:
Yep, that's the way to get L2 block number, by using precompile call. I'll mention that it can be used